hoomd/version.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) 2009-2019 The Regents of the University of Michigan
+# This file is part of the HOOMD-blue project, released under the BSD 3-Clause
+# License.
+
+"""Version and build information.
+
+Attributes:
+    compile_date (str): The date this build was compiled.
+
+    compile_flags (str): Human readable summary of compilation flags.
+
+    cxx_compiler (str): Name and version of the C++ compiler used to build
+        HOOMD.
+
+    enable_gpu (bool): ``True`` when this build supports GPUs.
+
+    enable_mpi (bool): ``True`` when this build supports MPI parallel runs.
+
+    enable_tbb (bool): ``True`` when this build supports TBB threads.
+
+    git_branch (str):  Name of the git branch used when compiling this build.
+
+    git_sha1 (str):  SHA1 of the git commit used when compiling this build.
+
+    gpu_api_version (str): The GPU API version this build was compiled against.
+
+    gpu_platform (str): Name of the GPU platform this build was compiled
+        against.
+
+    install_dir (str): The installation directory.
+
+    source_dir (str): The source directory.
+"""
+from hoomd import _hoomd
+
+version = "2.9.2"
```

Comment:
Even though it isn't PEP8 compliant, I feel we should probably follow the tradition of using `__version__` as that is what most people will expect.

Reply:
I disagree. Not only is it not PEP8 compliant, `__*__` are defined as system-defined names in the Python language specification: https://docs.python.org/3/reference/lexical_analysis.html#reserved-classes-of-identifiers
```
System-defined names, informally known as “dunder” names. These names are defined by the interpreter and its implementation (including the standard library). Current system names are discussed in the Special method names section and elsewhere. More will likely be defined in future versions of Python. Any use of __*__ names, in any context, that does not follow explicitly documented use, is subject to breakage without warning.
```
`__version__`, while commonly used, is not universal and it is not documented in the Python language specification or documentation: https://docs.python.org/3/search.html?q=__version__&check_keywords=yes&area=default
Users should expect to find the HOOMD-blue version in the attribute we document in the sphinx docs.

In the Python standard library, for example, a few packages provide `__version__`, (such as `http.server` and `json`), however these are being removed over time (for example, see the Python 3.5 release notes https://docs.python.org/3/whatsnew/3.5.html where `__version__` was dropped from the `email` package). Of these, only `http.server.__version__` is actually documented. Interestingly, `curses.__version__` is documented, but doesn't exist in Python 3.8. `curses`, `ensurepip`, `marshal`, and others provide a `version` attribute, function (which sometimes provides a string and sometimes a tuple).

The `version` module in HOOMD follows that of `numpy` e.g. `numpy.version.git_revision`

Reply:
Okay, well I am not opinionated enough to argue against this.